src/components/ScrollToTop/ScrollToTop.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Zoom, useScrollTrigger } from '@material-ui/core';
+import { useStyles } from './style';
+
+export const ScrollToTop = (props) => {
+  const { children, scrollStep, delayInMS } = props;
+  const classes = useStyles();
+  let intervalId;
+
+  const trigger = useScrollTrigger({
+    disableHysteresis: true,
+    threshold: 100,
+  });
+
+  const scrollToTopStep = () => {
+    if (window.pageYOffset === 0) {
+      clearInterval(intervalId);
+    }
+    window.scroll({
+      left: 0,
+      top: window.pageYOffset - scrollStep,
+      behavior: 'smooth',
+    });
+  };
+
+  const handleClick = () => {
+    intervalId = setInterval(scrollToTopStep, delayInMS);
```

Comment:
Can you explain why you are using delay here?

Reply:
I use it to call the function `scrollToTopStep` (which scrolls up `scrollStep` amount of pixel only once) repeatedly within delay time period. In other words, it helps me set up a loop that stops when `pageYOffset = 0`